codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -319,44 +338,56 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
      * @param in            the {@link ByteBuf} from which to read data
      * @param out           the {@link List} to which decoded messages should be added
      */
-    protected void callDecode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) {
-        try {
-            while (in.isReadable()) {
-                int outSize = out.size();
-                int oldInputLength = in.readableBytes();
+    protected boolean callDecode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) {
```

Comment:
we can no do this as this break the API. Let me come up with a PR that not do this. @ninja- ok ?